src/AspNet.Security.OAuth.Alipay/AlipayCertificationUtil.cs

```diff
@@ -0,0 +1,105 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Buffers;
+using System.Globalization;
+using System.Numerics;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace AspNet.Security.OAuth.Alipay;
+
+/// <summary>
+/// https://github.com/alipay/alipay-sdk-net-all/blob/b482d75d322e740760f9230d2a3859090af642a7/v2/AlipaySDKNet.Standard/Util/AntCertificationUtil.cs
+/// </summary>
+internal static class AlipayCertificationUtil
+{
+    public static string GetCertSN(ReadOnlySpan<char> certContent)
+    {
+        using var cert = X509Certificate2.CreateFromPem(certContent);
+        return GetCertSN(cert);
+    }
+
+    public static string GetCertSN(X509Certificate2 cert)
+    {
+        var issuerDN = cert.Issuer.Replace(", ", ",", StringComparison.InvariantCulture);
+        var serialNumber = new BigInteger(cert.GetSerialNumber()).ToString(CultureInfo.InvariantCulture);
+
+        if (issuerDN.StartsWith("CN", StringComparison.InvariantCulture))
+        {
+            return CalculateMd5(issuerDN + serialNumber);
+        }
+
+        var attributes = issuerDN.Split(',');
+        Array.Reverse(attributes);
+        return CalculateMd5(string.Join(',', attributes) + serialNumber);
+    }
+
+    public static string GetRootCertSN(ReadOnlySpan<char> rootCertContent, string signType = "RSA2")
+    {
+        var rootCertSN = string.Join('_', GetRootCertSNCore(rootCertContent, signType));
+        return rootCertSN;
+    }
+
+    private static IEnumerable<string> GetRootCertSNCore(X509Certificate2Collection x509Certificates, string signType)
+    {
+        foreach (X509Certificate2 cert in x509Certificates)
+        {
+            var signatureAlgorithm = cert.SignatureAlgorithm.Value;
+            if (signatureAlgorithm != null)
+            {
+                if ((signType.StartsWith("RSA", StringComparison.InvariantCultureIgnoreCase) &&
+                    signatureAlgorithm.StartsWith("1.2.840.113549.1.1", StringComparison.InvariantCultureIgnoreCase)) ||
```

Comment:
Based on [this search](https://github.com/search?q=repo%3Adotnet%2Fruntime%20%221.2.840.113549.1.1%22&type=code), `"1.2.840.113549.1.1."` seems more correct?

Otherwise would it pick up values in `11-19`, `100-199`, `1000-1999` etc. that it shouldn't?